### PR TITLE
[FIX] Do not include inventory loss in ADU computation

### PR DESCRIPTION
**Pull request by sebalix**
Inventory loss moves should not be included in the ADU past/future computation (not relevant).

Ref. 1830

### CODE REVIEW

File: ddmrp/models/stock_buffer.py [hunk 1]
```diff
@@ -978,10 +978,14 @@ def _past_demand_estimate_domain(self, date_from, date_to, locations):
...
         self.ensure_one()
+        # Exclude inventory loss moves
+        inventory_locations = self.env["stock.location"].search(
+            [("usage", "=", "inventory")]
+        )
         return [
             ("state", "=", "done"),
             ("location_id", "in", locations.ids),
-            ("location_dest_id", "not in", locations.ids),
+            ("location_dest_id", "not in", (locations | inventory_locations).ids),
```
Comment (simahawk) [inline ddmrp/models/stock_buffer.py]: can't we save a search by using `("location_dest_id.usage", "in", (...allowed_values...))`?

Comment (sebalix) [reply ddmrp/models/stock_buffer.py]: Good point, but using a dotted syntax in a search domain doesn't generate another query too?

**Review by jgrandguillaume — APPROVED** [review]
Functionally speaking, this is ok ! Thx

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
Makes sense! +1

File: ddmrp/models/stock_buffer.py (hunk 1 above)
Comment (sebalix) [reply ddmrp/models/stock_buffer.py]: After checking the execution query plan of both requests with @guewen , the one with `location_dest_id.usage` could be better at the end with lots of moves, it needs confirmation but I prefer to change it this way for now as it also makes the code clearer about the intent.

**Review by guewen — APPROVED** [review]

File: ddmrp/models/stock_buffer.py [hunk 2]
```diff
@@ -982,6 +982,7 @@ def _past_moves_domain(self, date_from, date_to, locations):
             ("state", "=", "done"),
             ("location_id", "in", locations.ids),
             ("location_dest_id", "not in", locations.ids),
+            ("location_dest_id.usage", "!=", "inventory"),
```
Comment (LoisRForgeFlow) [inline ddmrp/models/stock_buffer.py:985]: Could you include an extra domain based on a company setting to be able to include scraps, something like:
```python
if not self.env.company.ddmrp_adu_calc_include_scrap:
    domain.append(("location_id.scrap_location", "=" , False))
```

cc @jgrandguillaume I have checked the option to use a production location as a scrap location but Odoo only lets you use internal or inventory loss location to scrap.

**Comment by LoisRForgeFlow** [comment]
@sebalix Thanks! LGTM, could you fixup the second commit?

**Comment by sebalix** [comment]
@LoisRForgeFlow done

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM, tested locally.

**Pull request merged**